extensions/wikia/CommunityPage/models/CommunityPageSpecialUsersModel.class.php

```diff
@@ -0,0 +1,371 @@
+<?php
+
+class CommunityPageSpecialUsersModel {
+	const TOP_CONTRIB_MCACHE_KEY = 'community_page_top_contrib';
+	const FIRST_REV_MCACHE_KEY = 'community_page_first_revision';
+	const GLOBAL_BOTS_MCACHE_KEY = 'community_page_global_bots';
+	const ALL_MEMBERS_MCACHE_KEY = 'community_page_all_members';
+	const MEMBER_COUNT_MCACHE_KEY = 'community_member_count';
+	const RECENTLY_JOINED_MCACHE_KEY = 'community_page_recently_joined';
+	const CURR_USER_CONTRIBUTIONS_MCACHE_KEY = 'community_page_current_user_contributions';
+
+	private $wikiService;
+
+	private function getWikiService() {
+		return $this->wikiService;
+	}
+
+	public function __construct() {
+		$this->wikiService = new WikiService();
+	}
+
+	/**
+	 * Get the date a member made their first edit to a wiki.
+	 * Membership is defined as having edited in the last two years
+	 *
+	 * @param User $user
+	 * @return Mixed|null
+	 */
+	public function getFirstRevisionDate( User $user ) {
+		if ( $user->isAnon() ) {
+			return null;
+		}
+
+		$data = WikiaDataAccess::cache(
+			wfMemcKey( self::FIRST_REV_MCACHE_KEY ),
+			WikiaResponse::CACHE_STANDARD,
+			function () use ($user) {
+				$db = wfGetDB( DB_SLAVE );
+
+				$sqlData = ( new WikiaSQL() )
+					->SELECT( '*' )
+					->FROM ( 'wikia_user_properties' )
+					->WHERE ( 'wup_property' )->EQUAL_TO( 'firstContributionTimestamp' )
+					->AND_ ( 'wup_user' )->EQUAL_TO( $user->getID() )
+					->AND_ ( 'wup_value > DATE_SUB(now(), INTERVAL 2 YEAR)' )
+					->ORDER_BY( 'wup_value DESC' )
+					->run( $db, function ( ResultWrapper $result ) {
+						$out = [];
+						while ( $row = $result->fetchRow() ) {
```

Comment:
Rather than doing your own loop, just use WikiaSQL's `runLoop` rather than `run`, i.e. something like (untested):
```php
$sqlData = ( new WikiaSQL() )
	// ...query built here...
	->runLoop( $db, function ( &$sqlData, $row ) {
		$sqlData[] = [
			'wup_value' => $row->wup_value,
		];
	} );
```